### PR TITLE
feat: Replaced AddressBook based NetworkInfo implementations

### LINKED ISSUE #15635: Replace `AddressBook`-based implementations of `NetworkInfo`
Currently the main `NetworkInfo` implementations derive from the `AddressBook`.

We want to replace these with two implementations,
  1. _(For genesis)_ A `GenesisNetworkInfo` that throws `UnsupportedOperationException` on any method (e.g., `selfNodeInfo()`) not needed for non-genesis migrations and is otherwise derived from a `Roster` loaded from disk and injected via `ServicesMain`. This implementation should use a default `AccountID` for each node.
  2. _(For restart)_ A `StateNetworkInfo` that implements all methods from state---specifically, the `Roster` in platform state and the `AddressBookService`'s `Node` store for the node's fee collection account ids.

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Fixes #15635 

- Added `GenesisNetworkInfo` that is used when on genesis trigger. This networkInfo is constructed from the `config.txt` on disk. It uses nodeAccountId as (nodeId + 3)
- Changed `NetworkInfoImpl` to `StateNetworkInfo` . This is constructed from the `Roster` in PlatformState and `NodeState`. `NodeState` is only used to lookup the `nodeAccountId` in the networkInfo. This is updated each time a node update transaction is handled successfully.
- Removed `internalHostName`, `internalPort`, `externalHostName` and `externalPort` from `NodeInfo`. Instead used the list of `gossipEndPoints`

**Deleted Files:**
- Deleted `SelfNodeInfoImpl` and used `NodeInfoImpl` instead. Since the only difference was `hapiVersion` which can be obtained in other ways.
- Deleted `UnavailableLedgerIdNetworkInfo` which is no more needed. Used `GenesisNetworkInfo` instead
- Deleted `AbstractNetworkInfoImpl` since we don't have much common code between `GenesisNetworkInfo` and `StateNetworkInfo`

**Gradle Changes:**

- Since `Browser` run is deprecated, made the gradle run task to use `ServicesMain`. This avoid s to add all special conditions needed just supporting Browser

### CODE REVIEW

**Review by edward-swirldslabs — APPROVED** [review]

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 1]
```diff
@@ -522,7 +529,7 @@ public void onStateInitialized(
         }
     }
 
-    /**
+    /**act
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:532]: Intentional?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/OrderedServiceMigrator.java [hunk 2]
```diff
@@ -68,12 +68,11 @@ public List<StateChanges.Builder> doMigrations(
             @Nullable final SoftwareVersion previousVersion,
             @NonNull final SoftwareVersion currentVersion,
             @NonNull final Configuration config,
-            @NonNull final NetworkInfo networkInfo,
+            @Nullable final NetworkInfo genesisNetworkInfo,
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/OrderedServiceMigrator.java:71]: Isn't the ordered service migrator called in non-genesis cases as well? 

Hedera.java:
```
    * Initialization Step 2: Initialize the state. Either genesis or restart or reconnect or some other trigger.
    * Includes migration when needed.
```

File: platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/MigrationContext.java [hunk 3]
```diff
@@ -63,7 +63,8 @@ public interface MigrationContext {
      *
      * @return The {@link NetworkInfo} of the network at the time of migration.
      */
-    NetworkInfo networkInfo();
+    @Nullable
+    NetworkInfo genesisNetworkInfo();
```
Comment (mhess-swl) [inline platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/MigrationContext.java:67]: Again, isn't this also the network info used in reconnect (or other non-genesis) scenarios?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 4]
```diff
@@ -287,6 +289,11 @@ public final class Hedera implements SwirldMain, PlatformStatusChangeListener {
      */
     @Nullable
     private CompletableFuture<Bytes> initialStateHashFuture;
+    /**
+     * The roster used on genesis. This is set in {@link ServicesMain} on platform initialization.
+     * This is used to construct the {@link GenesisNetworkInfo} object.
+     */
+    private Roster genesisRoster;
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:296]: Only for genesis, or also for reconnect etc.?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip993/SystemFileExportsTest.java [hunk 5]
```diff
@@ -457,11 +454,6 @@ private static VisibleItemsValidator nodeDetailsExportValidator(
                             actualCertHash,
                             "node" + address.nodeId() + " has wrong cert hash");
 
-                    final var expectedAccountID = AccountID.newBuilder()
-                            .accountNum(address.nodeId() + ACCOUNT_ID_OFFSET)
-                            .build();
-                    assertEquals(expectedAccountID, address.nodeAccountId());
-
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip993/SystemFileExportsTest.java:464]: Why aren't these assertions needed anymore?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/DabEnabledUpgradeTest.java [hunk 6]
```diff
@@ -288,7 +288,6 @@ final Stream<DynamicTest> exportedAddressBookReflectsOnlyEditsBeforePrepareUpgra
                     // Now make some changes that should not be incorporated in this upgrade
                     nodeDelete("5"),
                     nodeDelete("2"),
-                    nodeUpdate("0").accountId(classicFeeCollectorIdLiteralFor(900)),
```
Comment (mhess-swl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/DabEnabledUpgradeTest.java:291]: Does this not work anymore?

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/standalone/TransactionExecutorsTest.java [hunk 7]
```diff
@@ -338,4 +348,27 @@ private Bytes resourceAsBytes(@NonNull final String loc) {
             throw new UncheckedIOException(e);
         }
     }
+
+    public static X509Certificate randomX509Certificate() {
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/standalone/TransactionExecutorsTest.java:352]: Can this test not use the method in `RosterRetrieverTest`?

File: hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/AppTestBase.java [hunk 8]
```diff
@@ -356,14 +338,14 @@ public App build() {
...
                     .toList();
-
-            final var platform = new FakePlatform(realSelfNodeInfo.nodeId(), new AddressBook(addresses));
-            final var networkInfo = new NetworkInfoImpl(realSelfNodeInfo, platform, configProvider);
-
+            final var addressBook = new AddressBook(addresses);
+            final var platform = new FakePlatform(realSelfNodeInfo.nodeId(), addressBook);
             final var initialState = new FakeState();
+            //            final var networkInfo = new NetworkInfoImpl(initialState,
+            //                    selfNodeInfo.nodeId(), configProvider);
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/AppTestBase.java:347]: remove?

File: hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/state/FakeState.java [hunk 9]
```diff
@@ -66,7 +66,11 @@ public class FakeState implements State {
      */
     public FakeState addService(@NonNull final String serviceName, @NonNull final Map<String, ?> dataSources) {
         final var serviceStates = this.states.computeIfAbsent(serviceName, k -> new ConcurrentHashMap<>());
-        serviceStates.putAll(dataSources);
+        dataSources.forEach((k, b) -> {
+            if (!serviceStates.containsKey(k)) {
+                serviceStates.put(k, b);
+            }
+        });
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/state/FakeState.java:73]: This slightly changes the behavior, yes? Instead of putting all data – i.e. overwriting any data already there – this only puts data that is new. Just making sure that's intended?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java [hunk 10]
```diff
@@ -857,10 +870,6 @@ private void initializeDagger(
             initialStateHashFuture = new CompletableFuture<>();
             notifications.register(ReconnectCompleteListener.class, new ReadReconnectStartingStateHash(notifications));
         }
-        if (initialStateHashFuture == null) {
-            logger.warn("Starting from Browser is deprecated. Setting initial start hash to empty hash.");
-            initialStateHashFuture = completedFuture(Bytes.wrap(new byte[48]));
-        }
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:863]: How is this related to the network info changes?

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/GenesisNetworkInfo.java [hunk 11]
```diff
@@ -0,0 +1,134 @@
...
+    /**
+     * Builds a node info from a roster entry from the given roster.
+     *
+     * @param entry The roster entry
+     * @return The node info
+     */
+    private NodeInfo fromRosterEntry(RosterEntry entry) {
+        return new NodeInfoImpl(
+                entry.nodeId(),
+                asAccount(entry.nodeId() + 3),
```
Comment (mhess-swl) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/GenesisNetworkInfo.java:112]: A comment explaining why the `+ 3` is necessary may be helpful

File: platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/MigrationContext.java (hunk 3 above)
Comment (Neeharika-Sompalli) [reply platform-sdk/swirlds-state-api/src/main/java/com/swirlds/state/spi/MigrationContext.java:67]: The `networkInfo` is essentially set on `MigrationContextImpl` and is used in `AddressBookService` and `TokenService` only on `GenesisTrigger`

File: hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/standalone/TransactionExecutorsTest.java (hunk 7 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/test/java/com/hedera/node/app/workflows/standalone/TransactionExecutorsTest.java:352]: Not accessible, so had to copy it.

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/DabEnabledUpgradeTest.java (hunk 6 above)
Comment (Neeharika-Sompalli) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/regression/system/DabEnabledUpgradeTest.java:291]: same as above

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip993/SystemFileExportsTest.java (hunk 5 above)
Comment (Neeharika-Sompalli) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/hip993/SystemFileExportsTest.java:464]: The test was doing node AccountId update after prepare upgrade is scheduled. But after upgrade the state is same as the one in prepare upgrade. This is not possible unless we maintain candidate roster, active roster and future roster in state. Those are future items. 
Since node accountId update is disabled by default, I removed that in the test.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java (hunk 4 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:296]: Removed this.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/OrderedServiceMigrator.java (hunk 2 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/services/OrderedServiceMigrator.java:71]: True. But the networkInfo is used only in genesis case. So I wanted to clarify and renamed it.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java (hunk 1 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:532]: No typo.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java [hunk 12]
```diff
@@ -232,24 +232,27 @@ public static void main(final String... args) throws Exception {
...
                 .withConfiguration(configuration)
                 .withAddressBook(addressBook)
-                .withRoster(createRoster(addressBook))
+                .withRoster(roster)
                 .withKeysAndCerts(keysAndCerts);
 
         hedera.setInitialStateHash(stateHash);
+        // FUTURE: This is needed because platform still doesn't have an API to read
+        // Roster set on it. This will be removed once platform has an API to read Roster.
+        hedera.setRoster(roster);
```
[the remark this answers is not in the document]
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/ServicesMain.java:255]: yeah agreed. We will just duplicate `createRoster`, but thats fine I guess. Changed it. Thanks @mhess-swl

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/GenesisNetworkInfo.java (hunk 11 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/info/GenesisNetworkInfo.java:112]: yes, since its only used in genesis, we  can use nodeId + 3 as accountId.

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java (hunk 10 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java:863]: We had to add all this to support Browser, like added in the log. 
Since I removed supporting Browser in gradle run, removed this

File: hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/state/FakeState.java (hunk 9 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/state/FakeState.java:73]: This was a bug surfaced while testing. We should add any state only when state is not present.

File: hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/AppTestBase.java (hunk 8 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-app/src/testFixtures/java/com/hedera/node/app/fixtures/AppTestBase.java:347]: Removed. Thanks

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java [hunk 13]
```diff
@@ -93,14 +93,14 @@ public void migrate(@NonNull final MigrationContext ctx) {
...
         for (final var nodeInfo : addressBook) {
             final var nodeBuilder = Node.newBuilder()
                     .nodeId(nodeInfo.nodeId())
                     .accountId(nodeInfo.accountId())
-                    .description(nodeInfo.selfName())
-                    .gossipEndpoint(List.of(
-                            endpointFor(nodeInfo.internalHostName(), nodeInfo.internalPort()),
-                            endpointFor(nodeInfo.externalHostName(), nodeInfo.externalPort())))
+                    // Default node description hard coded to the values used currently
+                    .description("node" + (nodeInfo.nodeId() + 1))
```
Comment (anthony-swirldslabs) [inline hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java:102]: Minor: note that we've introduced an utility `RosterUtils.formatNodeName(long id)` to ensure the consistency and avoid any code duplication. You may want to use it here to make sure that the entire code base always uses the exact same format for node names, everywhere.

**Review by anthony-swirldslabs — APPROVED** [review]

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli !

File: hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java (hunk 13 above)
Comment (Neeharika-Sompalli) [reply hedera-node/hedera-addressbook-service-impl/src/main/java/com/hedera/node/app/service/addressbook/impl/schemas/V053AddressBookSchema.java]: yes. Changed it. Thanks @anthony-swirldslabs

**Review by anthony-swirldslabs — APPROVED** [review]

File: hedera-node/hedera-addressbook-service-impl/build.gradle.kts [hunk 14]
```diff
@@ -21,6 +21,8 @@ plugins {
 
 description = "Default Hedera AddressBook Service Implementation"
 
+dependencies { implementation(project(":swirlds-platform-core")) }
+
```
Comment (mhess-swl) [inline hedera-node/hedera-addressbook-service-impl/build.gradle.kts:25]: Is this necessary? I thought we were able to do this just by depending on the right classes in `module-info.java`

Comment (mhess-swl) [reply hedera-node/hedera-addressbook-service-impl/build.gradle.kts:25]: Verified, we should be able to get rid of this. We can do it in a separate PR though since the checks are passing

**Review by mhess-swl — APPROVED** [review]
LGTM, tyvm @Neeharika-Sompalli!

**Review by dalvizu — APPROVED** [review]

**Review by edward-swirldslabs — APPROVED** [review]
